src/visu.f90

```diff
@@ -576,22 +581,23 @@ subroutine write_field(f1, pathname, filename, num, skip_ibm, flush)
   end subroutine write_field
 
   function gen_snapshotname(pathname, varname, num, ext)
-    character(len=*), intent(in) :: pathname, varname, num, ext
+    character(len=*), intent(in) :: pathname, varname, ext
+    integer, intent(in) :: num
+    character(len=:), allocatable :: gen_snapshotname
 #ifndef ADIOS2
-    character(len=(len(pathname) + 1 + len(varname) + 1 + len(num) + 1 + len(ext))) :: gen_snapshotname
-    write(gen_snapshotname, "(A)") gen_filename(pathname, varname, num, ext)
+    gen_snapshotname = gen_filename(pathname, varname, num, ext)
 #else
-    character(len=(len(varname) + 1 + len(num) + 1 + len(ext))) :: gen_snapshotname
-    write(gen_snapshotname, "(A)") varname//'-'//num//'.'//ext
+    gen_snapshotname = varname//'-'//int_to_str(num)//'.'//ext
 #endif
   end function gen_snapshotname
   
   function gen_filename(pathname, varname, num, ext)
 
-    character(len=*), intent(in) :: pathname, varname, num, ext
+    character(len=*), intent(in) :: pathname, varname, ext
+    integer, intent(in) :: num
 #ifndef ADIOS2
-    character(len=(len(pathname) + 1 + len(varname) + 1 + len(num) + 1 + len(ext))) :: gen_filename
-    write(gen_filename, "(A)") pathname//'/'//varname//'-'//num//'.'//ext
+    character(len=:), allocatable :: gen_filename
+    gen_filename = pathname//'/'//varname//'-'//int_to_str(num)//'.'//ext
```

Comment:
This statement allocates the character array "gen_filename" ?

Reply:
Yes - allocatable intrinsic types are (re)allocated on assignment (F2003).

Reply:
Nice